src/net/RTP/RTPSession.cs

```diff
@@ -1029,7 +1029,20 @@ public virtual SetDescriptionResultEnum SetRemoteDescription(SdpType sdpType, SD
                         SDPAudioVideoMediaFormat.SortMediaCapability(capabilities, currentMediaStream.LocalTrack?.Capabilities);
 
                         currentMediaStream.RemoteTrack.Capabilities = capabilities;
-                        currentMediaStream.LocalTrack.Capabilities = capabilities;
+
+                        // Keep the local track's RTP event capability
+                        SDPAudioVideoMediaFormat localRTPEventCapabilities;
+                        try
+                        {
+                            localRTPEventCapabilities = currentMediaStream.LocalTrack.Capabilities.First(x => x.Name().ToLower() == SDP.TELEPHONE_EVENT_ATTRIBUTE);
+                        }

```

Comment:
This looks a bit clunky. Why not use the linq .Any method to see if  the capabilty exists rather than catching the exception from .First?

Reply:
Addressed - let me know if this needs tightening-up some more!